cumulusci/tasks/bulkdata/generate_mapping.py

```diff
@@ -355,3 +390,36 @@ def _is_lookup_to_included_object(self, field):
         return field["type"] == "reference" and self._are_lookup_targets_in_operation(
             field
         )
+
+
+class FieldData:
+    nillable: bool
+
+    def __init__(self, describe_data: dict):
+        self.nillable = describe_data.get("nillable")
+
+    def __eq__(self, other: "FieldData"):
+        return self.__dict__ == other.__dict__
+
+
+def is_standard(obj: str):
+    return not obj.endswith("__c")
+
+
+def soft_dependency(sobj, target_obj: str, field_data: FieldData):
+    return field_data.nillable or (sobj == "Account" and is_standard(target_obj))
```

Comment:
I think this is pleonastic here. Account doesn't have any non-nillable standard parent relationships other than `OwnerId`, which we ignore.

I previously wanted to propose cycle breaking by making Account go first over any standard object that is acting as its child, through either a standard or custom relationship. Maybe that is more complexity than is needed. Looks like a reasonable way to do something similar would be to do a check in `find_free_object()` that returns Account if it is present in the list. That seems likely to me to produce optimal results the vast majority of the time. 

What do you think?

Reply:
Do you have specific cases in mind? 

If I delete the special case and run the code below on a clean org, it generates 800 steps without asking anything. It generates 54 "After" statements and only 2 of them are "after: Insert Account". One is an Account self-reference. The other is NetworkSelfRegistration. So it seems to guess where to put Account okay without any special code. The code does use lexicography sorting so Account ends up early just because of the English alphabet.

When I add some special code it inserts Account earlier but the total number of after: statements increases from 54 to 56.

So is there any particular use-case that I should test with demonstrate the benefit of boosting Account early in the process?

Reply:
@prescod  I'm fine with the pragmatic approach with no special code. My suggestion was not informed by data - thanks for doing the experiment.